McpMobileToolchainServiceTest: default Android profile and node version resolve empty/null

## Summary

Two tests in `shaft-mcp/src/test/java/com/shaft/mcp/McpMobileToolchainServiceTest.java` fail (reproducibly, even run in isolation — not test-order dependent) on a clean `origin/main` checkout at commit 3b145f4bfd (PR #3333, "Guard SHAFT.Properties initialization in McpMobileToolchainService").

This was discovered while fixing an unrelated compile break in the same file/PR (see #3341), and is being filed separately since it's a behavioral bug in production code rather than a compile issue.

## Failures

### 1. `defaultAndroidProposalUsesPortableCacheAndHostAbi`
```
org.opentest4j.AssertionFailedError: expected: <pixel_8> but was: <>
	at com.shaft.mcp.McpMobileToolchainServiceTest.defaultAndroidProposalUsesPortableCacheAndHostAbi(McpMobileToolchainServiceTest.java:124)
```
Reproduces even when run as the only test in the class:
```
mvn -pl shaft-mcp -am test -Dtest=McpMobileToolchainServiceTest#defaultAndroidProposalUsesPortableCacheAndHostAbi
```

### 2. `ensureAppiumInstallsPinnedPackagesWithCurrentCliSyntax`
```
java.lang.IllegalStateException: Portable Node.js could not be downloaded.
	at com.shaft.mcp.McpMobileToolchainService.downloadPortableNode(McpMobileToolchainService.java:472)
	at com.shaft.mcp.McpMobileToolchainService.ensureNpm(McpMobileToolchainService.java:420)
	at com.shaft.mcp.McpMobileToolchainService.ensureAppium(McpMobileToolchainService.java:301)
	at com.shaft.mcp.McpMobileToolchainServiceTest.ensureAppiumInstallsPinnedPackagesWithCurrentCliSyntax(McpMobileToolchainServiceTest.java:181)
Caused by: java.io.IOException: Download failed with HTTP 404: https://nodejs.org/dist/vnull/node-vnull-win-x64.zip
```
Note the `vnull` in the URL — `nodeLtsVersion()` is resolving to `null`.

## Likely root cause

`McpMobileToolchainService.ensurePropertiesInitialized()` (shaft-mcp/src/main/java/com/shaft/mcp/McpMobileToolchainService.java:747-776, added by PR #3333) caches `SHAFT.Properties.internal.*` values into **static** fields, guarded by a **static** `propertiesInitialized` boolean:

```java
private static void ensurePropertiesInitialized() {
    if (!propertiesInitialized) {
        synchronized (McpMobileToolchainService.class) {
            if (!propertiesInitialized) {
                try {
                    if (SHAFT.Properties.internal == null) {
                        PropertiesHelper.initialize();
                    }
                    cachedNodeLtsVersion = SHAFT.Properties.internal.nodeLtsVersion();
                    ...
                    cachedAndroidEmulatorDeviceProfile = SHAFT.Properties.internal.androidEmulatorDeviceProfile();
                    ...
                    propertiesInitialized = true;
                } catch (RuntimeException exception) { ... }
            }
        }
    }
}
```

In the `shaft-mcp` test classpath/environment, `SHAFT.Properties.internal.nodeLtsVersion()` and `.androidEmulatorDeviceProfile()` are resolving to `null`/empty instead of their expected defaults (`pixel_8`, an actual LTS version string), suggesting either:
- `PropertiesHelper.initialize()` isn't finding/loading the default properties file(s) that supply these values in the `shaft-mcp` module's test context, or
- the static caching model itself is fragile — once any code path in the JVM/test-fork calls `ensurePropertiesInitialized()` before properties are fully configured, the (possibly empty) values are cached for the lifetime of the fork and never refreshed.

## Suggested investigation

- Check whether `shaft-mcp/src/test/resources` (or its dependency on `shaft-engine`) actually has the default properties defining `androidEmulatorDeviceProfile` and `nodeLtsVersion` on the test classpath.
- Reconsider caching `SHAFT.Properties.internal.*` values in `static` fields at all — since these are meant to reflect current SHAFT.Properties state, static caching across the JVM/test fork seems like the wrong lifetime for values that should probably be read fresh (or cached per-instance) instead.

## Reproduction
```
mvn -pl shaft-mcp -am test -Dtest=McpMobileToolchainServiceTest
```
(reproduces on a clean `origin/main` checkout, unrelated to any other in-flight changes)

--- a/shaft-capture/src/test/java/com/shaft/capture/generate/CaptureGeneratorTest.java
+++ b/shaft-capture/src/test/java/com/shaft/capture/generate/CaptureGeneratorTest.java
@@ -463,7 +463,7 @@ class CaptureGeneratorTest {
         String source = Files.readString(result.sourcePath());
         assertTrue(source.contains(
                 "driver.element().assertThat(SHAFT.GUI.Locator.inputField(\"Username\")).attribute(\"autocomplete\")"
-                        + ".isEqualTo(requiredData(\"username\")).perform();"));
+                        + ".isEqualTo(requiredData(\"username\"));"));
     }
 
     @Test

--- a/shaft-mcp/src/main/java/com/shaft/mcp/McpMobileToolchainService.java
+++ b/shaft-mcp/src/main/java/com/shaft/mcp/McpMobileToolchainService.java
@@ -61,7 +61,6 @@ final class McpMobileToolchainService {
     McpMobileToolchainService() {
         this(McpProcessRunner.system(), System.getenv(), McpRuntimePaths.applicationDataRoot().resolve("tools"),
                 System.getProperty("os.name", ""), System.getProperty("os.arch", ""));
-        ensurePropertiesInitialized();
     }
 
     McpMobileToolchainService(
@@ -70,6 +69,7 @@ final class McpMobileToolchainService {
             Path toolRoot,
             String osName,
             String osArch) {
+        ensurePropertiesInitialized();
         this.runner = runner;
         this.environment = environment == null ? Map.of() : Map.copyOf(environment);
         this.toolRoot = toolRoot.toAbsolutePath().normalize();

